File: 2216-delete-the-middle-node-of-a-linked-list/delete-the-middle-node-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        if (head == nullptr || head->next == nullptr){
            return nullptr;
        }
        ListNode dummy = ListNode();
        dummy.next = head;
        ListNode* slow = &dummy;//head;
        ListNode* fast = &dummy;//head;
        while(fast->next != nullptr && fast->next->next != nullptr){
            slow = slow->next;
            fast = fast->next->next;
            // could have just counted number of jumps ig
        }
        slow->next = slow->next->next;
        slow = slow->next;
        return dummy.next;

    }
};

File: 2216-delete-the-middle-node-of-a-linked-list/delete-the-middle-node-of-a-linked-list.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function deleteMiddle(head: ListNode | null): ListNode | null {
    if (head ===null){
        return null;
    }
    let dummy = new ListNode(0,head);
    let slow = dummy
    let fast = dummy
    while (fast.next && fast.next.next){
        slow = slow.next
        fast = fast.next.next
    }
    slow.next = slow.next.next
    return dummy.next
};